container-search/src/main/java/com/yahoo/search/dispatch/CloseableInvoker.java

```diff
@@ -21,8 +20,8 @@ public abstract class CloseableInvoker implements Closeable {
     private RequestDuration duration;
 
     public void teardown(BiConsumer<Boolean, RequestDuration> teardown) {
-        this.teardown = teardown;
-        this.duration = new RequestDuration();
+        this.teardown = this.teardown == null ? teardown : this.teardown.andThen(teardown);
```

Comment:
Consider using `CompletableFuture`. Does the exception semantics for `andThen()` fit here?

Reply:
Yes, trying not to change too much at once. There shouldn't be any exceptions here, all we do is update a few numbers, but ... yes, we could handle a potential exception, indeed.